src/DataAPI.jl

```diff
@@ -288,34 +288,76 @@ using a `sink` function to materialize the table.
 function allcombinations end
 
 """
-    metadata(x, [key])
+    metadata(x)
 
 Return metadata associated with object `x` as an `AbstractDict{String}` object
 (or an object implementing the same interface).
 
-If the optional `key` argument is passed, return metadata
-associated with the indicated key (this feature can be used, for example,
-to return metadata attached to a given column in a Tables.jl table).
-
 Note that some systems, like Arrow.jl, might assume that metadata values are
 also `String`.
 """
 metadata(::T) where {T} =
-    throw(ArgumentError("Metadata is currently not supported for values of type $T"))
-metadata(::T, ::Any) where {T} =
-    throw(ArgumentError("Key metadata is currently not supported for values of type $T"))
+    throw(ArgumentError("Objects of type $T do not support metadata"))
+
+"""
+    colmetadata(table, column)
+
+Return metadata associated with column `column` of Tables.jl table `table` as an
+`AbstractDict{String}` object (or an object implementing the same interface).
+
+Note that some systems, like Arrow.jl, might assume that metadata values are
+also `String`.
+"""
+colmetadata(::T, ::Any) where {T} =
+    throw(ArgumentError("Objects of type $T do not support column metadata"))
+
+"""
+    colmetadata(table)
+
+Return metadata associated with columns of Tables.jl table `table` as an
+`AbstractDict{<:Any, <:AbstractString{String}}` object (or an object implementing
+the same interface).
 
+The returned dictionary must contain columns `column` for which
+`hascolmetadata(table, column)` returns `true` and a value associated to such
+key must be `colmetadata(table, column)`.
 """
-    hasmetadata(x, [key])
+colmetadata(::T) where {T} =
+    throw(ArgumentError("Objects of type $T do not support column metadata"))
```

Comment:
Maybe leave this out for now? It's just a convenience function that's easy to replicate by calling `Tables.columnnames` and `colmetadata`, and AFAIK we don't need it currently.

Reply:
We can leave it out. The only difference is performance. If you get `colmetadata(table)` to be `false` you know that NO column metadata is present (so you do not need to check individual columns)